lib/onelogin/ruby-saml/response.rb

```diff
@@ -51,6 +53,21 @@ def initialize(response, options = {})
 
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
+
+        if assertion_encrypted?
+          if @settings.nil? || !@settings.get_sp_key
+            validation_error('An EncryptedAssertion found and no SP private key found on the settings to decrypt it. Be sure you provided the :settings parameter at the initialize method')
+          end
+
+          # Marshal at Ruby 1.8.7 throw an Exception
+          if RUBY_VERSION < "1.9"
+            @decrypted_document = XMLSecurity::SignedDocument.new(@response, @errors)
+          else
+            @decrypted_document = Marshal.load(Marshal.dump(@document))
+          end
+
+          @decrypted_document = document_with_decrypted_assertion
```

Comment:
why do we need to store `@decrypted_document` in the `if` above and then write to the same variable here again?